src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -1544,6 +1544,40 @@ public static IResourceBuilder<T> WithCommand<T>(
         return builder.WithAnnotation(new ResourceCommandAnnotation(name, displayName, commandOptions.UpdateState ?? (c => ResourceCommandState.Enabled), executeCommand, commandOptions.Description, commandOptions.Parameter, commandOptions.ConfirmationMessage, commandOptions.IconName, commandOptions.IconVariant, commandOptions.IsHighlighted));
     }
 
+    /// <summary>
+    /// Adds an executable command to the resource builder with the specified name, display name, and command string.
+    /// </summary>
+    /// <typeparam name="T">The type of the resource.</typeparam>
+    /// <param name="builder">The resource builder to which the command will be added.</param>
+    /// <param name="name">The unique name of the command.</param>
+    /// <param name="displayName">The display name of the command, shown in the dashboard.</param>
+    /// <param name="command">The command string to be executed.</param>
+    /// <param name="commandOptions">Optional settings for the command, such as description and icon.</param>
+    /// <returns>The resource builder, allowing for method chaining.</returns>
+    public static IResourceBuilder<T> WithExecCommand<T>(
```

Comment:
I'm not sure about ExecCommand as the name.

Reply:
as per requirement in the issue (see https://github.com/dotnet/aspire/issues/10301), it should be with the name.
It allows to add multiple commands per name to the same container resource. for example
```
builder
  .AddResource(containerResource)
  .WithExecCommand("listfiles", ...)
  .WithExecCommand("deletefiles", ...)
```

Why are you not sure about such API? It is basically a copy-cat from `WithCommand`